packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -1038,9 +1038,18 @@ export class Configuration {
           continue;
 
         for (const userPluginEntry of data.plugins) {
-          const userProvidedPath = typeof userPluginEntry !== `string`
-            ? userPluginEntry.path
-            : userPluginEntry;
+          let userProvidedPath: string
+          if (typeof userPluginEntry !== `string`) {
+            if (userPluginEntry.enabled === `false`) {
+              // all built-in modules have already been required,
+              // however none of their functionality has been initialized yet
+              plugins.delete(userPluginEntry.spec);
```

Comment:
This seemed like the best way to go about it without rearchitecting the whole plugin loading system, because core plugins are put into the `Set` earlier in the file. The part of the code is evaluated, however since build-in plugins don't contain side-effects, this effectively disables them.

I considered moving the core plugin later to after this `data.plugins ` iteration, but then we'd change order of plugin execution, which might break things, so I didn't want to risk it.